test/Npgsql.Tests/ConnectionTests.cs

```diff
@@ -205,7 +205,7 @@ public void ProvidePasswordCallbackDelegateIsUsed()
             string ProvidePasswordCallback(string host, int port, string database, string username)
             {
                 getPasswordDelegateWasCalled = true;
-                return goodPassword;
+                return goodPassword!;
```

Comment:
This test assumes there's a good password in the default connection string - that's OK but we can just make it explicit... Let's make an explicit assert above that connString.Password is non-null, and make goodPassword non-nullable (so a bang wouldn't be needed here).

Reply:
This bang should be removed now, right?

Reply:
No, because the connection string builder stores it as nullable.

Reply:
OK, not very important in any case. One day NUnit will add the proper annotation to Assume.That and we won't need the bang anymore.